DevEdu.API/DevEdu.Business.Tests/CourseServiceTests.cs

```diff
@@ -20,11 +20,136 @@ public class CourseServiceTests
+
+            var sut = new CourseService(_courseRepositoryMock.Object,
+                                        _topicRepositoryMock.Object,
+                                        _taskRepositoryMock.Object,
+                                        _materialRepositoryMock.Object);
+            //When
+            var actualCourse = sut.AddCourse(courseDto);
+
+            //Than
+            Assert.AreEqual(courseId, actualCourse);
+            _courseRepositoryMock.Verify(x => x.AddCourse(courseDto), Times.Once());
+        }
+
+        [Test]
+        public void DeleteCourse_IntCourseId_DeleteCourse()
+        {
+            //Given
+            var courseDto = CourseData.GetCourseDto();
+            var courseId = CourseData.CourseId;
+
+            _courseRepositoryMock.Setup(x => x.DeleteCourse(courseId));
+
+            var sut = new CourseService(_courseRepositoryMock.Object,
+                                        _topicRepositoryMock.Object,
+                                        _taskRepositoryMock.Object,
+                                        _materialRepositoryMock.Object);
+            //When
+            sut.DeleteCourse(courseId);
+
+            //Than
+            _courseRepositoryMock.Verify(x => x.DeleteCourse(courseId), Times.Once());
+        }
+
+        [Test]
+        public void GetCourseById_IntCourseId_ReturnCourseWithGroups()
+        {
+            //Given
+            var courseDto = CourseData.GetCourseDto();
+            var courseId = CourseData.CourseId;
+            _courseRepositoryMock.Setup(x => x.GetCourse(courseId)).Returns(courseDto);
+
+            var sut = new CourseService(_courseRepositoryMock.Object,
+                                        _topicRepositoryMock.Object,
+                                        _taskRepositoryMock.Object,
+                                        _materialRepositoryMock.Object);
+            //When
+            var dto = sut.GetCourse(courseId);
+
+            //Than
+            Assert.AreEqual(courseDto, dto);
+            _courseRepositoryMock.Verify(x => x.GetCourse(1));
+        }
+        [Test]
+        public void GetCourseById_IntCourseId_ReturnCourseWithGroups_Topics_Materials_Tasks()
+        {
+            //Given
+            var courseDto = CourseData.GetCourseDto();
+            var courseId = CourseData.CourseId;
+            _courseRepositoryMock.Setup(x => x.GetCourse(courseId)).Returns(courseDto);
+
+            var sut = new CourseService(_courseRepositoryMock.Object,
+                                        _topicRepositoryMock.Object,
+                                        _taskRepositoryMock.Object,
+                                        _materialRepositoryMock.Object);
+            //When
+            var dto = sut.GetFullCourseInfo(courseId);
+
+            //Than
+            Assert.AreEqual(courseDto, dto);
+            _courseRepositoryMock.Verify(x => x.GetCourse(courseId));
```

Comment:
please add verification for those methods:
![image](https://user-images.githubusercontent.com/18329725/127709478-065addeb-5df5-4010-992c-caf612ed8705.png)

Reply:
+